app/scripts/lib/resolver.js

```diff
@@ -0,0 +1,71 @@
+const namehash = require('eth-ens-namehash')
+const multihash = require('multihashes')
+const HttpProvider = require('ethjs-provider-http')
+const Eth = require('ethjs-query')
+const EthContract = require('ethjs-contract')
+const registrarAbi = require('./contracts/registrar')
+const resolverAbi = require('./contracts/resolver')
+
+function ens (name, provider) {
+  const eth = new Eth(new HttpProvider(getProvider(provider.type)))
+  const hash = namehash.hash(name)
+  const contract = new EthContract(eth)
+  const Registrar = contract(registrarAbi).at(getRegistrar(provider.type))
+  return new Promise((resolve, reject) => {
+    if (provider.type === 'mainnet' || provider.type === 'ropsten') {
+      Registrar.resolver(hash).then((address) => {
+        if (address === '0x0000000000000000000000000000000000000000') {
+          reject(null)
+        } else {
+          const Resolver = contract(resolverAbi).at(address['0'])
+          return Resolver.content(hash)
+        }
+      }).then((contentHash) => {
+        if (contentHash['0'] === '0x0000000000000000000000000000000000000000000000000000000000000000') reject(null)
+        if (contentHash.ret !== '0x') {
+          const hex = contentHash['0'].substring(2)
+          const buf = multihash.fromHexString(hex)
+          resolve(multihash.toB58String(multihash.encode(buf, 'sha2-256')))
+        } else {
+          reject(null)
+        }
+      })
+    } else {
+      return reject('unsupport')
+    }
+  })
+}
+
+function getProvider (type) {
+  switch (type) {
+    case 'mainnet':
+      return 'https://mainnet.infura.io/'
```

Comment:
I would rather we use the user's provider, provided by MetaMask, to reduce our number of blockchain connections, but for this feature, I can see the benefit of this simplicity, there aren't other chains w/ the ENS resolver anyway, this _could_ be optimized later...